tox.ini

```diff
@@ -77,7 +77,7 @@ setenv =
          COVERAGE_PROCESS_START=.coveragerc
 commands =
          # NOTE: 'run with "py.test --keep-tempdir -s -v integration/" to debug failures'
-         py.test --coverage -v {posargs:integration}
+         py.test --timeout=1800 --coverage -v {posargs:integration}
```

Comment:
Oh this looks like a nice global timeout.  But from the magnitude ... I guess it is a timeout for the *whole suite*?  Conveniently `pytest --help` doesn't tell me:

```
   --timeout=TIMEOUT     Timeout in seconds before dumping the stacks. Default
                         is 0 which means no timeout.
```

It seems like a per-test timeout would be better than a global timeout.

Reply:
I believe it is in fact per-test.